### PR TITLE
Simcom 

### PR DESCRIPTION
**Pull request by karthi012**
add simcom modem driver abstraction

### CODE REVIEW

**Comment by DineshDK03** [comment]
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_espressif | https://github.com/zephyrproject-rtos/hal_espressif/commit/8e5da13712012d2cc480f7abc699235352153e05 | https://github.com/zephyrproject-rtos/hal_espressif/commit/0b5aa325aead1f36e104615f966e171cf4d36538 (zephyr) | [zephyrproject-rtos/hal_espressif@8e5da137..0b5aa325](https://github.com/zephyrproject-rtos/hal_espressif/compare/8e5da13712012d2cc480f7abc699235352153e05..0b5aa325aead1f36e104615f966e171cf4d36538) |
| mcuboot | https://github.com/zephyrproject-rtos/mcuboot/commit/6902abba270c0fbcbe8ee3bb56fe39bc9acc2774 | https://github.com/zephyrproject-rtos/mcuboot/commit/1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe (main,upstream-sync) | [zephyrproject-rtos/mcuboot@6902abba..1558e7ab](https://github.com/zephyrproject-rtos/mcuboot/compare/6902abba270c0fbcbe8ee3bb56fe39bc9acc2774..1558e7ab0aadb4eac11f03befb5ccd3fa3f0aafe) |
| openthread | https://github.com/zephyrproject-rtos/openthread/commit/7bdcf8a5d49838ce6e3e227e2780e4f12c461330 | https://github.com/zephyrproject-rtos/openthread/commit/25942910ab57b32f73d9c56d8c6cf07c7cf98c49 (main) | [zephyrproject-rtos/openthread@7bdcf8a5..25942910](https://github.com/zephyrproject-rtos/openthread/compare/7bdcf8a5d49838ce6e3e227e2780e4f12c461330..25942910ab57b32f73d9c56d8c6cf07c7cf98c49) |
| sof | https://github.com/zephyrproject-rtos/sof/commit/ac71254757478f79aa88807c168e88aee70a330b | https://github.com/zephyrproject-rtos/sof/commit/ffbf9c2a6ea2930b0ac7e3a37c7cd7f5c417d090 (zephyr) | [zephyrproject-rtos/sof@ac712547..ffbf9c2a](https://github.com/zephyrproject-rtos/sof/compare/ac71254757478f79aa88807c168e88aee70a330b..ffbf9c2a6ea2930b0ac7e3a37c7cd7f5c417d090) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

File: drivers/modem/simcom-y7080.c [hunk 1]
```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2023 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT simcom
```
Comment (ssekar15) [inline drivers/modem/simcom-y7080.c:7]: driver is not compatible for all the simcom, pls restrict only for 7080 and y7080. e.g drivers/sensor/tmp108/tmp108.c

File: drivers/modem/simcom-y7080.c [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	if (hdop == NULL) {
+		hdop = "0";
+	}
+
+	gnss_data->lat = get_float_from_str(lat);
+	gnss_data->lat_dir = lat_dir[0];
+	gnss_data->lon = get_float_from_str(lon) ;
+	gnss_data->lon_dir = lon_dir[0];
+	if (alt) {
+		gnss_data->alt =  get_float_from_str(alt);;
```
Comment (ssekar15) [inline drivers/modem/simcom-y7080.c:129]: nit ;;

**Comment by ssekar15** [comment]
still draft or ready for review ??

File: drivers/modem/simcom-xxxx.h [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+	uint8_t mac_addr[6];
+	/*
+	 * Uart interface of the modem.
+	 */
+	struct modem_iface_uart_data iface_data;
+	uint8_t iface_rb_buf[MDM_MAX_DATA_LENGTH];
+	/*
+	 * Modem specific Init
+	 */
+	simcom_mdm_init init;
```
Comment (ssekar15) [inline drivers/modem/simcom-xxxx.h:96]: never used...

**Review by ssekar15 — COMMENTED** [review]
I just stopped review..... cleanup code with checkpatch

Do a self review of your code before ask for review

**Pull request closed without merging**